### PR TITLE
Fix issue #20 by adding line buffering

### PR DESCRIPTION
**Pull request by olystretch**
Adding line buffering when opening log file ensures you only read complete lines. Fixes issue #20

### CODE REVIEW

File: pygtail/core.py [hunk 1]
```diff
@@ -168,9 +168,9 @@ def _filehandle(self):
         if not self._fh or self._is_closed():
             filename = self._rotated_logfile or self.filename
             if filename.endswith('.gz'):
-                self._fh = gzip.open(filename, 'r')
+                self._fh = gzip.open(filename, "r", 1)
```
Comment (bgreenlee) [inline pygtail/core.py]: 1 here means compression level, not buffering

**Comment by bgreenlee** [comment]
Looks good. Remove the change to gzip.open and I'll merge.

**Comment by olystretch** [comment]
D'oh! Good call. I only use Pygtail on live logs, so I didn't catch the mistake.

**Comment by bgreenlee** [comment]
Thank you!

**Pull request merged**